src/server/graphics/software_cursor.h

```diff
@@ -58,6 +58,7 @@ class SoftwareCursor : public Cursor
     std::shared_ptr<input::Scene> const scene;
     MirPixelFormat const format;
     std::mutex guard;
+    std::mutex show_hide_guard; ///< keep this locked when adding or removing renderable from the scene
     std::shared_ptr<detail::CursorRenderable> renderable;
     bool visible;
```

Comment:
Am I right in thinking the mutex guards `renderable` and `visible`? There's nothing in the naming, comments or layout to make this explicit.

The class has two mutextes: `guard` and `show_hide_guard` and without understanding all the code there's no way to know how they are to be used.